src/button-group/icon-button-item.tsx

```diff
@@ -0,0 +1,52 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { forwardRef } from 'react';
+import { ButtonGroupProps } from './interfaces.js';
+import { InternalButton } from '../button/internal.js';
+import { ButtonProps } from '../button/interfaces.js';
+import { ClickDetail } from '../internal/events/index.js';
+import { isDevelopment } from '../internal/is-development.js';
+import { warnOnce } from '@cloudscape-design/component-toolkit/internal';
+import styles from './styles.css.js';
+import testUtilStyles from './test-classes/styles.css.js';
+import clsx from 'clsx';
+
+const IconButtonItem = forwardRef(
+  (
+    {
+      item,
+      onItemClick,
+    }: {
+      item: ButtonGroupProps.IconButton;
+      onItemClick?: (event: CustomEvent<ClickDetail>) => void;
+    },
+    ref: React.Ref<ButtonProps.Ref>
+  ) => {
+    const hasIcon = item.iconName || item.iconUrl || item.iconSvg;
+
+    if (isDevelopment && !hasIcon) {
```

Comment:
I believe isDevelopment is already checked by warnOnce

Reply:
We should also add a unit test for the warning

Reply:
How do you test warnOnce?

Reply:
We mock it and assert calls, see example: https://github.com/cloudscape-design/components/blob/2c3bfbe12bfe8773f12896a09664f911d468bd85/src/expandable-section/__tests__/expandable-section.test.tsx#L295